pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go

```diff
@@ -160,6 +163,7 @@ func (grm *nestedPendingOperations) isOperationExists(
 	podName types.UniquePodName) (bool, int) {
 
 	// If volumeName is empty, operation can be executed concurrently
+	// TODO-P0: I think we want to prevent GenerateVolumesAreAttachedFunc operating concurrently
```

Comment:
When I work on it, I was thinking all these checking operation could be executed concurrently with other types of volume operations such as attaching, detaching. Even if there is race condition, there is no harm to set volume as detached. So the next round reconciler will mark as attached if volume is already attached. To prevent multiple operations for checking on the same volume, I will think what is the best way to handle it.